plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -416,6 +420,21 @@ public void checkCanRenameMaterializedView(ConnectorSecurityContext context, Sch
         }
     }
 
+    @Override
+    public void checkCanGrantExecuteFunctionPrivilege(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName functionName, TrinoPrincipal grantee, boolean grantOption)
+    {
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                if (!isAdmin(context)) {
+                    String granteeAsString = format("%s '%s'", grantee.getType().name().toLowerCase(ENGLISH), grantee.getName());
+                    denyGrantExecuteFunctionPrivilege(functionName.toString(), Identity.ofUser(context.getIdentity().getUser()), granteeAsString);
+                }
+        }
+        throw new UnsupportedOperationException("Unsupported function kind: " + functionKind);
```

Comment:
No return. UnsupportedOperationException will be thrown fro admin

Reply:
`denyGrantExecuteFunctionPrivilege` throws

Reply:
..throws for not admin.

Reply:
Fixed